### ISSUE TITLE
Home-manager flake fails

### ISSUE THREAD

**Issue by helmha** [librephoenix/nixos-config#1]
Hello,

Saw your video on YT about NixOS and flakes and wanted to try out your .dotfiles in a VM per your instructions at install.org. 

Building the #system works but when I run **home-manager switch --flake .#user** it fails after a moment outputting the following:

`testuser@nixos:~/.dotfiles/ > home-manager switch --flake .#user                                                                                                                                                   
error:
       … while evaluating a branch condition

         at /nix/store/q2y6pk2zk3qdhsv3kjg7kywq2izvhsf3-nixpkgs-patched/lib/lists.nix:57:9:

           56|       fold' = n:
           57|         if n == len
             |         ^
           58|         then nul

       … while calling the 'length' builtin

         at /nix/store/q2y6pk2zk3qdhsv3kjg7kywq2izvhsf3-nixpkgs-patched/lib/lists.nix:55:13:

           54|     let
           55|       len = length list;
             |             ^
           56|       fold' = n:

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: value is a string while a Boolean was expected
testuser@nixos:~/.dotfiles/ > `

Channels used are:
```
home-manager https://github.com/nix-community/home-manager/archive/master.tar.gz
nixpkgs https://nixos.org/channels/nixpkgs-unstable

```

Any idea what might be the cause?

[edit]
Errm, for some reason it works when I change the profile from personal to homelab
[/edit]

**Comment by librephoenix** [comment]
@helmha Thanks for letting me know!

Hmmm...  Would you be able to post results of:  `home-manager switch --flake .#user --show-trace`

From the error, it looks like it may have something to do with my patched nixpkgs, but that should be affecting all profiles (homelab included).

**Comment by librephoenix** [comment]
I just did a quick reinstall inside of a VM as well and I couldn't reproduce that error, but here's a few things it might be:

- Is the `username` variable in flake.nix set to the name of your user (testuser)? I'm assuming you already discovered that (since you were changing the profile as well), but I guess that could be the cause of the issue?
- The theming code (from the themes directory) auto-downloads wallpapers from the internet, but the build will fail entirely if the wallpaper isn't found at the specified link.  The uwunicorn theme background is in this state at the moment (-̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥᷄◞ω◟-̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥̥᷅ ) so changing the `theme` variable to be named a different theme (found in the themes directory) could fix this

Other than that, I'm not exactly sure?  You could also try running
```
nix-channel --update
nix-shell '<home-manager>' -A install
```
to make sure home-manager is set to the correct (unstable) version before it starts managing itself.

If none of those things work though, perhaps the full trace will show us the real problem :)

**Comment by helmha** [comment]
I changed the 'editor' in flake.nix from emacs to nano, and that's what's causing all the trouble. Leaving it unchanged everything works okay.  Didn't notice theme error because I had changed it.

**Comment by librephoenix** [comment]
Aha!  That was the issue all along!  My if-else statement (for the spawnEditor variable) was wrong and caused that error if the editor variable was set to nano, vim or nvim.  Should be fixed in the latest commit [8700cb8](https://github.com/librephoenix/nixos-config/commit/8700cb8779af65ba86c823bb971f06854137907f)

Thanks for helping me find that mistake!

I'll mark this as closed since this fixes the original issue.
